src/averageAges.js

```diff
@@ -15,12 +15,18 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm');
+  const agesOfMen = century
+    ? (men.filter(man => Math.ceil(man.died / 100) === century)
+      .map(man => man.died - man.born))
+    : men.map(man => man.died - man.born);
```

Comment:
```suggestion
  const men = people.filter(person => 
    century
      ? Math.ceil(person.died / 100) === century && person.sex === 'm')
      : person.sex === 'm'
```

You could make it more short and filter men directly according to the `century`